tests/Sabberworm/CSS/ParserTest.php

```diff
@@ -435,4 +438,70 @@ function parsedStructureForFile($sFileName, $oSettings = null) {
 		return $oParser->parse();
 	}
 
+	/**
+	 * @depends testFiles
+	 */
+	function testLineNumbersParsing() {
+		$oDoc = $this->parsedStructureForFile('line-numbers');
+		// array key is the expected line number
+		$aExpected = array(
+			1 => array('Sabberworm\CSS\Property\Charset'),
+			3 => array('Sabberworm\CSS\Property\CSSNamespace'),
+			5 => array('Sabberworm\CSS\RuleSet\AtRuleSet'),
+			11 => array('Sabberworm\CSS\RuleSet\DeclarationBlock'),
+			// Line Numbers of the inner declaration blocks
+			17 => array('Sabberworm\CSS\CSSList\KeyFrame', 18, 20),
+			23 => array('Sabberworm\CSS\Property\Import'),
+			25 => array('Sabberworm\CSS\RuleSet\DeclarationBlock')
+		);
+
+		$aActual = array();
+		foreach ($oDoc->getContents() as $oContent) {
+			$aActual[$oContent->getLineNo()] = array(get_class($oContent));
+			if ($oContent instanceof KeyFrame) {
+				foreach ($oContent->getContents() as $block) {
+					$aActual[$oContent->getLineNo()][] = $block->getLineNo();
+				}
+			}
+		}
+
+		$aUrlExpected = array(7, 26); // expected line numbers
+		$aUrlActual = array();
+		foreach ($oDoc->getAllValues() as $oValue) {
+			if ($oValue instanceof URL) {
+				$aUrlActual[] = $oValue->getLineNo();
+			}
+		}
+
+		// Checking for the multiline color rule lines 27-31
+		$aExpectedColorLines = array(28, 29, 30);
+		$aDeclBlocks = $oDoc->getAllDeclarationBlocks();
+		// Choose the 2nd one
+		$oDeclBlock = $aDeclBlocks[1];
+		$aRules = $oDeclBlock->getRules();
+		// Choose the 2nd one
+		$oColor = $aRules[1]->getValue();
+		$this->assertEquals($aRules[1]->getLineNo(), 27);
+
+		foreach ($oColor->getColor() as $oSize) {
+			$aActualColorLines[] = $oSize->getLineNo();
+		}
+
+		$this->assertEquals($aExpectedColorLines, $aActualColorLines);
+		$this->assertEquals($aUrlExpected, $aUrlActual);
+		$this->assertEquals($aExpected, $aActual);
+	}
+
+	/**
+	 * @expectedException Sabberworm\CSS\Parsing\UnexpectedTokenException
+	 */
+	public function testUnexpectedTokenExceptionLineNo() {
+		$oParser = new Parser("\ntest: 1;", Settings::create()->beStrict());
+		try {
+			$oParser->parse();
+		} catch (UnexpectedTokenException $e) {
```

Comment:
Either use the full class namespace `Sabberworm\CSS\Parsing\UnexpectedTokenException` or `use Sabberworm\CSS\Parsing\UnexpectedTokenException` at the top of the file.